app/Voucher.php

```diff
@@ -63,6 +64,46 @@ class Voucher extends Model
         'sponsor_id' => ['numeric', 'required', 'exists:sponsors,id'],
     ];
 
+
+    /**
+     * Voucher can clean it's codes.
+     * @param array $codes
+     * @return array
+     */
+    public static function cleanCodes(array $codes)
```

Comment:
I guess you are assuming future may hold more bad chars? otherwise would strip spaces be more accurate name?

Reply:
Indeed, it would not suprise me if non-space codes or more mangling is required in the future.